weave/workers/weave_scorer.py

```diff
@@ -0,0 +1,445 @@
+import asyncio
+import inspect
+import logging
+import os
+from datetime import datetime
+from typing import Any, Optional
+
+import sentry_sdk
+from confluent_kafka import KafkaError, Message
+from tenacity import (
+    before_log,
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_fixed,
+)
+
+# This import is used to register built-in scorers so they can be deserialized from the DB
+import weave.scorers  # noqa: F401
+from weave.flow.monitor import Monitor
+from weave.flow.scorer import Scorer, get_scorer_attributes
+from weave.trace.box import box
+from weave.trace.objectify import maybe_objectify
+from weave.trace.op import _default_on_input_handler
+from weave.trace.serialization.serialize import to_json
+from weave.trace.weave_client import (
+    RUNNABLE_FEEDBACK_TYPE_PREFIX,
+    Call,
+    FeedbackCreateReq,
+    ObjectRef,
+    from_json,
+)
+from weave.trace_server import trace_server_interface as tsi
+from weave.trace_server.clickhouse_trace_server_batched import ClickHouseTraceServer
+from weave.trace_server.kafka import CALL_ENDED_TOPIC, KafkaConsumer
+from weave.trace_server.refs_internal import (
+    InternalCallRef,
+    InternalObjectRef,
+    parse_internal_uri,
+)
+from weave.trace_server.trace_server_interface import TraceServerInterface
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+_TRACE_SERVER: Optional[TraceServerInterface] = None
+
+
+def get_trace_server() -> TraceServerInterface:
+    global _TRACE_SERVER
+
+    if _TRACE_SERVER is None:
+        _TRACE_SERVER = ClickHouseTraceServer.from_env()
+
+    return _TRACE_SERVER
+
+
+# This should be cached to avoid hitting ClickHouse for each ended call.
+def get_active_monitors(
```

Comment:
I think you pretty much have to cache this now, else we are going to needlessly slam clickhouse. I'd recommend LRU with TTL of 1 min

Reply:
Do we have an LRU with TTL implementation somewhere already?
`functools.lru_cache` does not have TTL.